### PR TITLE
OSASINFRA-3319: implements required OpenStack credentials

### PR DESCRIPTION
**Pull request by EmilienM**
**What this PR does / why we need it**:

Refactor/clean and add missing credentials for the OpenStack platform.

### CODE REVIEW

**Comment by EmilienM** [comment]
@bryan-cox I think this is ready for review. This gave me a working cluster.
Let's prove it with:

/test e2e-openstack

Note: the job will fail because Ingress is not working (we'll work on that in 4.18, as it's big for us).

Please let me know if there are things that were not needed or not correct, I'll be happy to change.

**Comment by EmilienM** [comment]
CI is good, the openstack job isn't ready yet.

File: control-plane-operator/controllers/hostedcontrolplane/cloud/openstack/providerconfig.go [hunk 1]
```diff
@@ -7,51 +7,73 @@ import (
...
 	Provider            = "openstack"
-	cloudsSecretKey     = "clouds.yaml"
-	caSecretKey         = "cacert"
+	CloudsSecretKey     = "clouds.yaml"
+	CASecretKey         = "cacert"
 )
 
-// ReconcileCloudConfig reconciles as expected by Nodes Kubelet.
-func ReconcileCloudConfig(secret *corev1.Secret, hcp *hyperv1.HostedControlPlane, credentialsSecret *corev1.Secret, hasCACert bool) error {
+// ReconcileCloudConfig reconciles the cloud config secret.
```
Comment (MaysaMacedo) [inline control-plane-operator/controllers/hostedcontrolplane/cloud/openstack/providerconfig.go:19]: typo ReconcileCloudConfigSecret

**Review by MaysaMacedo — COMMENTED** [review]
/lgtm

I just had one minor comment

File: control-plane-operator/controllers/hostedcontrolplane/cloud/openstack/providerconfig.go (hunk 1 above)
Comment (EmilienM) [reply control-plane-operator/controllers/hostedcontrolplane/cloud/openstack/providerconfig.go]: done

**Comment by EmilienM** [comment]
Azure CI seems unstable today: https://prow.ci.openshift.org/job-history/gs/test-platform-results/pr-logs/directory/pull-ci-openshift-hypershift-main-e2e-azure
But I don't think this is related to this PR.

**Comment by bryan-cox** [comment]
/approve

Azure e2e is due to a CI issue with https://github.com/openshift/hypershift/pull/4408

**Pull request merged**